tree/ntuple/v7/src/RColumnElement.hxx

```diff
@@ -740,6 +742,173 @@ public:
    }
 };
 
+namespace Quantize {
+
+using Quantized_t = std::uint32_t;
+
+[[maybe_unused]] inline std::size_t LeadingZeroes(std::uint32_t x)
+{
+#ifdef _MSC_VER
+   unsigned long idx = 0;
+   _BitScanForward(&idx, x);
+   return static_cast<std::size_t>(idx);
+#else
+   return static_cast<std::size_t>(__builtin_clzl(x));
+#endif
+}
+
+[[maybe_unused]] inline std::size_t TrailingZeroes(std::uint32_t x)
+{
+#ifdef _MSC_VER
+   unsigned long idx = 0;
+   _BitScanReverse(&idx, x);
+   return static_cast<std::size_t>(idx);
+#else
+   return static_cast<std::size_t>(__builtin_ctzl(x));
+#endif
+}
+
+/// Converts the array `src` of `count` floating point numbers into an array of their quantized representations.
+/// Each element of `src` is assumed to be in the inclusive range [min, max].
+/// The quantized representation will consist of unsigned integers of at most `nQuantBits` (with `nQuantBits <= 8 *
+/// sizeof(Quantized_t)`). The unused bits are kept in the LSB of the quantized integers, to allow for easy bit packing
+/// of those integers via BitPacking::PackBits().
+/// \return The number of values in `src` that were found to be out of range (0 means all values were in range).
+template <typename T>
+int QuantizeReals(Quantized_t *dst, const T *src, std::size_t count, double min, double max, std::size_t nQuantBits)
+{
+   static_assert(std::is_floating_point_v<T>);
+   static_assert(sizeof(T) <= sizeof(double));
+   assert(1 <= nQuantBits && nQuantBits <= 8 * sizeof(Quantized_t));
+
+   const std::size_t quantMax = (1ull << nQuantBits) - 1;
+   const double scale = quantMax / (max - min);
+   const std::size_t unusedBits = sizeof(Quantized_t) * 8 - nQuantBits;
+
+   int nOutOfRange = 0;
+
+   for (std::size_t i = 0; i < count; ++i) {
+      T elem = src[i];
+
+      nOutOfRange += !(min <= elem && elem <= max);
+
+      double e = (elem - min) * scale;
+      Quantized_t q = static_cast<Quantized_t>(e + 0.5);
```

Comment:
Why split the two arithmetic operations?  If for clarify I would use:
```suggestion
      const double e = 0.5 + (elem - min) * scale;
      Quantized_t q = static_cast<Quantized_t>(e);
```

Reply:
In my first version I was flooring instead of rounding, then I realized rounding was the proper way to do it and I added it there. But yes, it makes more sense to have it on the line above.